src/main/java/net/tirasa/connid/bundles/ad/util/ADUtilities.java

```diff
@@ -528,41 +528,30 @@ public LdapEntry getEntryToBeUpdated(final String entryDN) {
         return obj;
     }
 
-    private String getEntryToBeUpdatedQuery(final Uid uid, final ObjectClass oclass) {
-        return connection.getSchema().getLdapUidAttribute(oclass) + "="
-                + (OBJECTGUID.equals(connection.getSchema().getLdapUidAttribute(oclass))
-                ? getEscapedGUID(uid.getUidValue())
-                : uid.getUidValue());
-    }
-
     public ConnectorObject getEntryToBeUpdated(final Uid uid, final ObjectClass oclass) {
         OperationOptionsBuilder builder = new OperationOptionsBuilder();
         builder.setAttributesToGet(Arrays.asList(UACCONTROL_ATTR, SDDL_ATTR, OBJECTSID, PRIMARYGROUPID));
 
-        LdapFilter filter = LdapFilter.forNativeFilter(getEntryToBeUpdatedQuery(uid, oclass));
+        final String filter = connection.getSchema().getLdapUidAttribute(oclass) + "=" + uid.getUidValue();
 
         LOG.ok("Searching for object of class {0} with filter {1}", oclass.getObjectClassValue(), filter);
 
-        final ConnectorObject obj = new LdapSearch(connection, oclass, filter, null, builder.build()) {
-
-            @Override
-            protected ConnectorObject createConnectorObject(final String baseDN,
-                    final SearchResult result,
-                    final Set<String> attrsToGet,
-                    final boolean emptyAttrWhenNotFound) {
-
-                try {
-                    // cannot use default createConnectorObject, since payload may contain Active Directory binary 
-                    // and/or special attributes
-                    return createMinimalConnectorObject(result.getNameInNamespace(),
-                            result.getAttributes(),
-                            attrsToGet,
-                            oclass);
-                } catch (NamingException e) {
-                    throw new ConnectorException("Error while creating connector object", e);
-                }
+        final ConnectorObject[] results = new ConnectorObject[] { null };
+        new ADSearch(connection, oclass, LdapFilter.forNativeFilter(filter), connectorObject -> {
+            results[0] = connectorObject;
+            return false;
+        }, builder.build()).execute((result, attrsToGet) -> {
+            try {
+                return createMinimalConnectorObject(result.getNameInNamespace(),
+                        result.getAttributes(),
+                        attrsToGet,
+                        oclass);
+            } catch (NamingException e) {
+                throw new ConnectorException(e);
```

Comment:
this used to be

```
throw new ConnectorException("Error while creating connector object", e);
```

any reason to change?

Reply:
no, let's keep the original one